boot/seal.go

```diff
@@ -717,30 +717,43 @@ func recoveryBootChainsForSystems(systems []string, modesForSystems map[string][
 				kernelRev = seedKernel.SideInfo.Revision.String()
 			}
 
-			recoveryBootChain, err := trbl.RecoveryBootChain(seedKernel.Path)
+			recoveryBootChains, err := trbl.RecoveryBootChains(seedKernel.Path)
 			if err != nil {
 				return err
 			}
 
+			foundChain := false
+
 			// get asset chains
-			assetChain, kbf, err := buildBootAssets(recoveryBootChain, modeenv)
-			if err != nil {
-				return err
+			for _, recoveryBootChain := range recoveryBootChains {
+				assetChain, kbf, err := buildBootAssets(recoveryBootChain, modeenv)
```

Comment:
buildBootAssets could take the TrustedAssets map and then BootFile could still carry paths?

Reply:
Done